components/seldon-request-logger/tests/integration_test.py

```diff
@@ -0,0 +1,265 @@
+        )
+    except AssertionError as e:
+        # For debugging. Left for future reference
+        logs = docker_services._docker_compose.execute("logs request_logger").decode("utf-8")
+        print(logs)
+        raise e
+
+
+@pytest.mark.parametrize("scenario, request_type", [
+    (metadata.kfserving_iris_batch, RequestType.REQUEST),
+    (metadata.kfserving_iris_batch, RequestType.RESPONSE),
+])
+def test_kfserving_iris_metadata(logger_service, elastic_service, scenario, request_type, docker_services,
+                                 kfserving_iris_deployment):
+    data, headers, expected_index, expected_docs = scenario(request_type)
+    try:
+        run_checks(
+            logger_service, elastic_service, data, headers, expected_index, expected_docs
+        )
+    except AssertionError as e:
+        # For debugging. Left for future reference
+        logs = docker_services._docker_compose.execute("logs request_logger").decode("utf-8")
+        print(logs)
+        raise e
+
+
+
+@pytest.mark.parametrize("scenario, request_type", [
+    (metadata.seldon_moviesentiment_text_no_batch, RequestType.REQUEST),
+    (metadata.seldon_moviesentiment_text_no_batch, RequestType.RESPONSE),
+    (metadata.seldon_moviesentiment_text_batch, RequestType.REQUEST),
+    (metadata.seldon_moviesentiment_text_batch, RequestType.RESPONSE),
+])
+def test_moviesentiment_text_metadata(logger_service, elastic_service, scenario, request_type, docker_services,
+                                    seldon_moviesentiment_deployment):
+    data, headers, expected_index, expected_docs = scenario(request_type)
+    try:
+        run_checks(
+            logger_service, elastic_service, data, headers, expected_index, expected_docs
+        )
+    except AssertionError as e:
+        # For debugging. Left for future reference
+        logs = docker_services._docker_compose.execute("logs request_logger").decode("utf-8")
+        print(logs)
+        raise e
+
+@pytest.mark.parametrize("scenario, request_type", [
+    (metadata.seldon_income_batch, RequestType.REQUEST),
+    (metadata.seldon_income_batch, RequestType.RESPONSE),
+    (metadata.seldon_income_batch, RequestType.REFERENCE_REQUEST),
+])
+def test_seldon_income_metadata(logger_service, elastic_service, scenario, request_type, docker_services,
+                                    seldon_income_deployment):
+    data, headers, expected_index, expected_docs = scenario(request_type)
+    try:
+        run_checks(
+            logger_service, elastic_service, data, headers, expected_index, expected_docs
+        )
+    except AssertionError as e:
+        # For debugging. Left for future reference
+        logs = docker_services._docker_compose.execute("logs request_logger").decode("utf-8")
+        print(logs)
+        raise e
+
+
+@pytest.mark.parametrize("scenario, request_type", [
+    # (metadata.seldon_cifar10_single, RequestType.REQUEST), # expected to fail
```

Comment:
There is a known issue with image type requests where only the first row is returned.

This is something that should be fixed in future